### PR TITLE
SAK-33832: edu-services prevent unique category naming error

### PR DESCRIPTION
**Pull request by bjones86**
https://jira.sakaiproject.org/browse/SAK-33832

Steps to reproduce:

1. Remove a category; do *not* press 'Save' yet
2. Add a new category, give it the exact same name as the one you just removed
3. Click the 'Save' button
4. Notice you get an error stating "Category names must be unique"

If you were to have clicked 'Save' in between the delete and the add, it would work just fine. However, this is not intuitive to end users, and it may seem as though it's a bug in the system.

This problem is half process/user error, half logic error; however we can make an improvement to avoid this, by reversing the logic in the back end method. Process the deletes first, *then* do the adds.

### CODE REVIEW

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 1]
```diff
@@ -3271,16 +3258,12 @@ private Double calculateCategoryScore(final String studentUuid, final Long categ
 			if(categoryId.longValue() != assignment.getCategory().getId().longValue()){
 				return true;
 			}
-
 			//remove if the assignment/graderecord doesn't meet the criteria for the calculation (rule 2-6)
-			if(assignment.getPointsPossible() == null || gradeRecord.getPointsEarned() == null || !assignment.isCounted() || !assignment.isReleased() || gradeRecord.getDroppedFromGrade()) {
```
Comment (steveswinsburg) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: prefer it if this wrapper was kept so it's a bit easier to see whats happening.

Comment (steveswinsburg) [reply edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: rest looks ok.

**Comment by bjones86** [comment]
Updated based on previous comment.

**Comment by steveswinsburg** [comment]
FYI rebasing makes it impossible to see what the changes are since the last comment, since there is no subsequent commit to look at 🤕.

**Comment by bjones86** [comment]
Yeah, it's a habit from before we had the 'Squash and merge' feature, so we always had to rebase. Old habits die hard...

**Pull request merged**